numpy/testing/nosetester.py

```diff
@@ -413,27 +402,55 @@ def test(self, label='fast', verbose=1, extra_argv=None,
             for warningtype in raise_warnings:
                 warnings.filterwarnings('error', category=warningtype)
             # Filter out annoying import messages.
-            warnings.filterwarnings('ignore', message='Not importing directory')
-            warnings.filterwarnings("ignore", message="numpy.dtype size changed")
-            warnings.filterwarnings("ignore", message="numpy.ufunc size changed")
-            warnings.filterwarnings("ignore", category=np.ModuleDeprecationWarning)
-            warnings.filterwarnings("ignore", category=FutureWarning)
```

Comment:
IIRC that filter hasn't been present for very long. It looks correct to me to remove it. ``FutureWarning`` should be treated similarly to ``DeprecationWarning`` I'd think.

If this is an issue in develop mode only, I think that's OK. We just have to fix scipy up, preferably right before merging this PR.